### PR TITLE
feat: add git tag struct and interface

### PR DESCRIPTION
**Pull request by l-qing**
# Changes
* feat: add git tag struct and interface

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [X] https://github.com/katanomi/spec/pull/141
- [X] Follows the [commit message standard](https://github.com/katanomi/spec/blob/main/CONTRIBUTING.md#commits)
- [X] Meets the [contributing guidelines](https://github.com/katanomi/spec/blob/main/CONTRIBUTING.md) (including
  functionality, content, code)
- [ ] Test cases with documentation and functionality works as expected using current and related github repos (MUST deploy and check)
- [X] Release notes block below has been filled in or deleted (only if no user facing changes)

### CODE REVIEW

File: plugin/route/gitrepotag.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	}
+}
+
+// Register route
+func (a *gitRepositoryTagLister) Register(ws *restful.WebService) {
+	projectParam := ws.PathParameter("project", "repository belong to project").DataType("string")
+	repositoryParam := ws.PathParameter("repository", "tag belong to repository")
+	ws.Route(
+		ws.GET("/projects/{project:*}/coderepositories/{repository}/tags").To(a.ListGitRepositoryTag).
+			Doc("GetGitRepoList").Param(projectParam).Param(repositoryParam).
```
Comment (airycanon) [inline plugin/route/gitrepotag.go:50]: doc 的名字是不是调整一下。

File: plugin/route/gitrepotag.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+}
+
+// Register route
+func (a *gitRepositoryTagGetter) Register(ws *restful.WebService) {
+	repositoryParam := ws.PathParameter("repository", "repository name")
+	projectParam := ws.PathParameter("project", "repository belong to project").DataType("string")
+	tagParam := ws.PathParameter("tag", "the name of the tag")
+	ws.Route(
+		ws.GET("/projects/{project:*}/coderepositories/{repository}/tags/{tag}").To(a.GetGitRepositoryTag).
+			Doc("GetGitRepo").Param(projectParam).Param(repositoryParam).Param(tagParam).
```
Comment (airycanon) [inline plugin/route/gitrepotag.go:93]: 同上。

File: plugin/route/gitrepotag.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+	repo := handlePathParamHasSlash(request.PathParameter("repository"))
+	tag := request.PathParameter("tag")
+	repoInfo, err := a.impl.GetGitRepositoryTag(request.Request.Context(),
+		metav1alpha1.GitRepositoryTagOption{
+			GitRepo: metav1alpha1.GitRepo{Repository: repo, Project: project},
+			Tag:     tag,
+		},
+	)
+	if err != nil {
+		if errors.IsNotFound(err) {
```
Comment (airycanon) [inline plugin/route/gitrepotag.go:111]: HandleError 也会处理 NotFound，不需要单独判断。

**Review by airycanon — CHANGES_REQUESTED** [review]

File: plugin/route/gitrepotag.go (hunk 1 above)
Comment (l-qing) [reply plugin/route/gitrepotag.go:50]: 👍

**Review by airycanon — APPROVED** [review]

**Pull request merged**